initializr-metadata/src/main/java/io/spring/initializr/metadata/BillOfMaterials.java

```diff
@@ -185,10 +186,25 @@ public void updateCompatibilityRange(VersionParser versionParser) {
 	 * additional BOMs to use, if any.
 	 * @param bootVersion the Spring Boot version
 	 * @return the bill of materials
+	 * @throws IllegalStateException if bom has mappings, none of which match the
+	 * requested version
 	 */
 	public BillOfMaterials resolve(Version bootVersion) {
+		return this.resolveSafe(bootVersion).orElseThrow(() -> new IllegalStateException(
+				"No suitable mapping was found for " + this + " and version " + bootVersion));
+	}
+
+	/**
+	 * Resolve this instance according to the specified Spring Boot {@link Version}.
+	 * Return a {@link BillOfMaterials} instance that holds the version, repositories and
+	 * additional BOMs to use, if any.
+	 * @param bootVersion the Spring Boot version
+	 * @return the bill of materials or empty if bom has mappings, none of which match the
+	 * requested version
+	 */
+	public Optional<BillOfMaterials> resolveSafe(Version bootVersion) {
```

Comment:
Not super keen on the `*Safe` name but I did not want to force all `resolve()` callers to change semantics - best name I could think of at the time.